app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt

```diff
@@ -19,16 +20,17 @@ import org.hisp.dhis.android.core.period.PeriodType
 import org.hisp.dhis.android.core.trackedentity.TrackedEntityDataValue
 import java.util.Date
 import java.util.Locale
-import javax.inject.Inject
 
-class ProgramEventMapper @Inject constructor(
+class ProgramEventMapper(
     val d2: D2,
     val periodUtils: DhisPeriodUtils,
+    val metadataIconProvider: MetadataIconProvider,
 ) {
 
     fun eventToEventViewModel(event: Event): EventViewModel {
         val programStage =
             d2.programModule().programStages().uid(event.programStage()).blockingGet()
+                ?: throw IllegalArgumentException()
```

Comment:
same here, maybe rather than throwing an exception we could display some kind of message informing the user that progam stage cannot be null